packages/toolbox-adk/README.md

```diff
@@ -10,18 +14,187 @@ pip install toolbox-adk
 
 ## Usage
 
+The primary entry point is the `ToolboxToolset`, which loads tools from a remote Toolbox server and adapts them for use with ADK agents.
+
+> [!NOTE]
+> The `ToolboxToolset` in this package mirrors the `ToolboxToolset` in the [`adk-python`](https://github.com/anubhav756/adk-python) package. The `adk-python` version is a shim that delegates all functionality to this implementation.
+
 ```python
 from toolbox_adk import ToolboxToolset, CredentialStrategy
+from google.adk.agents import Agent
 
-# Configure auth (e.g., Use the agent's identity)
-creds = CredentialStrategy.toolbox_identity()
+# 1. Configure Authentication Strategy
+# Use the agent's own identity (Workload Identity)
+creds = CredentialStrategy.workload_identity(target_audience="https://my-toolbox-service-url")
+
+# 2. Create the Toolset
+toolset = ToolboxToolset(
+    server_url="https://my-toolbox-service-url",
+    toolset_name="my-toolset", # Optional: Load specific toolset
+    credentials=creds
+)
+
+# 3. Use in your ADK Agent
+agent = Agent(tools=[toolset])
+```
+
+## Authentication Strategies
+
+The `toolbox-adk` package provides flexible authentication strategies to handle `Client-to-Server` authentication (securing the connection to the Toolbox server) and `User Identity` propagation (authenticating the user for specific tools).
+
+Use the `CredentialStrategy` factory methods to create your configuration.
+
+### Workload Identity (Recommended for Cloud Run / GKE)
+
+Uses the agent's environment credentials (ADC) to generate an OIDC ID token. This is the standard way for one service to authenticate to another on Google Cloud.
+
+```python
+# target_audience should match the URL of your Toolbox server
+creds = CredentialStrategy.workload_identity(target_audience="https://my-toolbox-service.run.app")
 
-# Create the toolset
 toolset = ToolboxToolset(
-    server_url="http://localhost:5000",
+    server_url="https://my-toolbox-service.run.app",
     credentials=creds
 )
+```
+
+### User Identity (3-Legged OAuth)
+
+Propagates the end-user's identity to the Toolbox. This is used when the tools themselves need to act on behalf of the user (e.g., accessing the user's Drive or Calendar).
+
+```python
+creds = CredentialStrategy.user_identity(
```

Comment:
Is this passed in to the ToolboxToolset the same as above? I worry about confusing the service to service with end user auth concepts